### PR TITLE
OAK-9707: Don't fail oak-run indexing on invalid data

### PR DESCRIPTION
**Pull request by tihom88**

### CODE REVIEW

File: oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java [hunk 1]
```diff
@@ -278,8 +279,15 @@ protected boolean indexTypeOrderedFields(Document doc, String pname, int tag, Pr
                 f = new SortedDocValuesField(name,
                         new BytesRef(property.getValue(Type.BOOLEAN).toString()));
             } else if (tag == Type.STRING.tag()) {
-                f = new SortedDocValuesField(name,
-                        new BytesRef(property.getValue(Type.STRING)));
+                String stringValue = property.getValue(Type.STRING);
+                BytesRef bytesRef;
```
Comment (thomasmueller) [inline oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java]: I would log the length, and simplify a bit, something like this (not tested):

    String value = property.getValue(Type.STRING);
    if (value.length() > STRING_PROPERTY_MAX_LENGTH) {
        log.warn("Truncating property {} for node {} as its length {} is > {}", 
                name, this.path, value.length(), STRING_PROPERTY_MAX_LENGTH);
        value = value.substring(0, STRING_PROPERTY_MAX_LENGTH));
    }
    f = new SortedDocValuesField(name, new BytesRef(value));

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneLargeStringPropertyTest.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        def.setProperty(TYPE_PROPERTY_NAME, LuceneIndexConstants.TYPE_LUCENE);
+        def.setProperty(REINDEX_PROPERTY_NAME, true);
+        def.setProperty(FulltextIndexConstants.FULL_TEXT_ENABLED, false);
+        def.setProperty(PropertyStates.createProperty(FulltextIndexConstants.INCLUDE_PROPERTY_NAMES, propNames, Type.STRINGS));
+        def.setProperty(LuceneIndexConstants.SAVE_DIR_LISTING, true);
+        return index.getChild(INDEX_DEFINITIONS_NAME).getChild(name);
+    }
+
+    @Test
+    public void truncateStringInCaseStringIsGreaterThanMAX_STRING_PROPERTY_LENGTH_AndOrderByIsConfigured() throws Exception {
```
Comment (thomasmueller) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneLargeStringPropertyTest.java]: I wouldn't use the constant name in the method name... what about:

    truncateStringInCaseStringIsGreaterThanMaxAndOrderByIsConfigured

**Review by thomasmueller — CHANGES_REQUESTED** [review]
Great! But we need to log the length... and some nitpicking.

File: oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java (hunk 1 above)
Comment (tihom88) [reply oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java]: Done

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneLargeStringPropertyTest.java (hunk 2 above)
Comment (tihom88) [reply oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneLargeStringPropertyTest.java]: done

**Review by thomasmueller — APPROVED** [review]

**Pull request merged**